pandas/core/algorithms.py

```diff
@@ -1502,7 +1504,12 @@ def take(
 # ------------ #
 
 
-def searchsorted(arr, value, side="left", sorter=None) -> np.ndarray:
+def searchsorted(
+    arr: ArrayLike,
+    value: NumpyValueArrayLike,
+    side: Literal["left", "right"] = "left",
+    sorter: NumpySorter = None,
+) -> np.ndarray:
```

Comment:
```suggestion
) -> np.ndarray | int:
```

although if we return the value from numpy unchanged, the return type is `np.intp`?
 
 docstring also needs updating?

Reply:
It is `np.intp` in next commit.  Will fix docstring.